infrastructure/cli/cli.go

```diff
@@ -33,15 +35,24 @@ type SnykCli struct {
 	authenticator snyk.AuthenticationService
 	errorReporter error_reporting.ErrorReporter
 	analytics     ux.Analytics
+	semaphore     chan int
 }
 
+const cliTimeout = 15 * time.Minute
+
 var Mutex = &sync.Mutex{}
 
 func NewExecutor(authenticator snyk.AuthenticationService, errorReporter error_reporting.ErrorReporter, analytics ux.Analytics) Executor {
+	concurrencyLimit := runtime.NumCPU()
```

Comment:
How do we limit to an only single spawned CLI **per product** in this solution?

Reply:
Yes, runtime.NumCPU() should not be used. One CLI process can easily take 5 cores on my machine, spawning several java executables and doing analysis itself. So we should limit concurrency to maybe a quarter of the cores at most.